weather.py

```diff
@@ -90,7 +176,37 @@ def generate_summary(weather_data):
     Returns:
         A string containing the summary information.
     """
-    pass
+    result = ""
+    no_days = len(weather_data)
+    result += str(no_days) + " Day Overview"
+    result += "\n"
+    low = []
+    high = []
+    for data in weather_data:
+        low.append(data[1])
+        high.append(data[2])
+    min_temp = format_temperature(convert_f_to_c(min(low)))
+    max_temp = format_temperature(convert_f_to_c(max(high)))
+    dates = []
+    for data in weather_data:
+        dates.append(data[0])
+    low_date = convert_date(dates[low.index(min(low))])
+    high_date = convert_date(dates[high.index(max(high))])
```

Comment:
Nice use of `min` `max` & `index`. Hint: there's also a function you created earlier that will do this for you :)